pkg/auditlogs/client_test.go

```diff
@@ -0,0 +1,316 @@
+			ExportsEndpoint: server.URL,
+		}
+		SetAPIKey("test")
+
+		body, err := CreateExport(context.TODO(), CreateExportOpts{})
+		require.Equal(t, body, CreateExportResponse{
+			Id: "test",
+		})
+		require.NoError(t, err)
+	})
+	t.Run("Call succeed with filters", func(t *testing.T) {
+		handlerFunc := func(w http.ResponseWriter, r *http.Request) {
+			var opts CreateExportOpts
+			dec := json.NewDecoder(r.Body)
+			dec.Decode(&opts)
+
+			require.Equal(t, opts.Actions[0], "create-user")
+			require.Equal(t, opts.Targets[0], "user")
+			require.Equal(t, opts.Actors, []string{"Jon", "Smith"})
+
+			body, _ := json.Marshal(CreateExportResponse{
+				Id: "test123",
+			})
+
+			w.Write(body)
+			w.WriteHeader(http.StatusOK)
+		}
+		server := httptest.NewServer(http.HandlerFunc(handlerFunc))
+		defer server.Close()
+
+		DefaultClient = &Client{
+			HTTPClient:      server.Client(),
+			ExportsEndpoint: server.URL,
+		}
+		SetAPIKey("test")
+
+		body, err := CreateExport(context.TODO(), CreateExportOpts{
+			Actions: []string{"create-user"},
+			Targets: []string{"user"},
+			Actors:  []string{"Jon", "Smith"},
+		})
+		require.Equal(t, body, CreateExportResponse{
+			Id: "test123",
+		})
+		require.NoError(t, err)
+	})
+	t.Run("401 requests returns an error", func(t *testing.T) {
+		handlerFunc := func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusUnauthorized)
+		}
+		server := httptest.NewServer(http.HandlerFunc(handlerFunc))
+		defer server.Close()
+
+		DefaultClient = &Client{
+			HTTPClient:      server.Client(),
+			ExportsEndpoint: server.URL,
+		}
+		SetAPIKey("test")
+
+		_, err := CreateExport(context.TODO(), CreateExportOpts{})
+		require.Error(t, err)
+	})
+}
+
+func TestGetExports(t *testing.T) {
+	t.Run("Call succeeds", func(t *testing.T) {
+		handlerFunc := func(w http.ResponseWriter, r *http.Request) {
+			body, _ := json.Marshal(GetExportResponse{
+				Id: "test",
+			})
+			w.WriteHeader(http.StatusCreated)
+			w.Write(body)
+		}
+		server := httptest.NewServer(http.HandlerFunc(handlerFunc))
+		defer server.Close()
+
+		DefaultClient = &Client{
+			HTTPClient:      server.Client(),
+			ExportsEndpoint: server.URL,
+		}
+		SetAPIKey("test")
+
+		body, err := GetExport(context.TODO(), GetExportOpts{})
+		require.Equal(t, body, GetExportResponse{
+			Id: "test",
+		})
+		require.NoError(t, err)
+	})
+	t.Run("401 requests returns an error", func(t *testing.T) {
+		handlerFunc := func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusUnauthorized)
+		}
+		server := httptest.NewServer(http.HandlerFunc(handlerFunc))
+		defer server.Close()
+
+		DefaultClient = &Client{
+			HTTPClient:      server.Client(),
+			ExportsEndpoint: server.URL,
+		}
+		SetAPIKey("test")
+
+		_, err := GetExport(context.TODO(), GetExportOpts{})
+		require.Error(t, err)
+	})
+}
+
+type defaultTestHandler struct {
+	header *http.Header
+}
+
+// Unit test to hit the prod api directly. Uncomment and provide an APIKey to
```

Comment:
Is this unit test necessary? I don't think any of our other tests have this, so I'm just wondering if something is unique here that requires this or if it was a specific ask (or if other unit tests should be improved)

Reply:
So I used this code as a reference:
https://github.com/workos/workos-go/blob/dfb61a3e3268f2e8843cfe87d30e4431b1685108/pkg/audittrail/client_test.go#L249-L269

It was actually useful when I tested live. I found a few errors and I was able to fix it.

Should we remove it?